pkg/ginutil/writeproblem.go

```diff
@@ -152,3 +152,94 @@ func WriteInvalidBindError(c *gin.Context, err error, detail string) {
 		Instance: c.Request.RequestURI,
 	})
 }
+
+// WriteAPIReadError uses WriteProblemError to write a 502 "Bad Gateway"
+// response with the type "/prob/api/unexpected-api-read-error".
+//
+// Meant to be used on unexpected error when reading data using the Wharf API.
+func WriteAPIReadError(c *gin.Context, err error, detail string) {
+	WriteProblemError(c, err, problem.Response{
+		Type: "/prob/provider/unexpected-api-read-error",
+		Title: "Unexpected API read error.",
+		Status: http.StatusBadGateway,
+		Detail: detail,
+	})
+}
+
+// WriteAPIWriteError uses WriteProblemError to write a 502 "Bad Gateway"
+// response with the type "/prob/api/unexpected-api-write-error".
+//
+// Meant to be used on unexpected error when writing data using the Wharf API.
+func WriteAPIWriteError(c *gin.Context, err error, detail string) {
+	WriteProblemError(c, err, problem.Response{
+		Type: "/prob/provider/unexpected-api-write-error",
+		Title: "Unexpected API write error.",
+		Status: http.StatusBadGateway,
+		Detail: detail,
+	})
+}
+
+// WriteResponseFormatError uses WriteProblemError to write a 502 "Bad Gateway"
+// response with the type "/prob/provider/unexpected-response-format".
+//
+// Meant to be used on unexpected error when the response format does not match our expectations.
+func WriteResponseFormatError(c *gin.Context, err error, detail string) {
+	WriteProblemError(c, err, problem.Response{
+		Type: "/prob/provider/unexpected-response-format",
+		Title: "Unexpected response format.",
+		Status: http.StatusBadGateway,
+		Detail: detail,
+	})
+}
+
+// WriteFetchBuildDefinitionError uses WriteProblemError to write a 400 "Bad Request"
+// response with the type "/prob/provider/fetch-build-definition".
+//
+// Meant to be used on error when fetching the build definition.
+func WriteFetchBuildDefinitionError(c *gin.Context, err error, detail string) {
+	WriteProblemError(c, err, problem.Response{
+		Type: "/prob/provider/fetch-build-definition",
+		Title: "Error fetching build definition.",
+		Status: http.StatusBadRequest,
+		Detail: detail,
+	})
+}
+
+// WriteComposingProviderDataError uses WriteProblemError to write a 400 "Bad Request"
+// response with the type "/prob/provider/composing-provider-data".
+//
+// Meant to be used on error when composing provider data.
+func WriteComposingProviderDataError(c *gin.Context, err error, detail string) {
+	WriteProblemError(c, err, problem.Response{
+		Type: "/prob/provider/composing-provider-data",
+		Title: "Error composing provider data.",
+		Status: http.StatusBadRequest,
+		Detail: detail,
+	})
+}
+
+// WriteTriggerError uses WriteProblemError to write a 502 "Bad Gateway"
+// response with the type "/prob/build/unexpected-trigger-error".
+//
+// Meant to be used on unexpected error during build process regarding a trigger.
+func WriteTriggerError(c *gin.Context, err error, detail string) {
+	WriteProblemError(c, err, problem.Response{
+		Type: "/prob/build/unexpected-trigger-error",
```

Comment:
This comment seems to just rephrase the problem type. Could this be elaborated? Also, this could perhaps also live in the `/prob/api-client/` problem namespace? This does fall nicely into the category, as we then have 3 main operations: read, write, & trigger.

```suggestion
// WriteTriggerError uses WriteProblemError to write a 502 "Bad Gateway"
// response with the type "/prob/api-client/unexpected-trigger-error".
//
// Meant to be used when unexpectedly failing to trigger a new build indirectly
// from an Wharf API client, such as from a Wharf provider plugin.
func WriteTriggerError(c *gin.Context, err error, detail string) {
	WriteProblemError(c, err, problem.Response{
		Type: "/prob/api-client/unexpected-trigger-error",
```